app/src/main/java/fr/neamar/kiss/loader/LoadShortcutsPojos.java

```diff
@@ -92,4 +92,19 @@ private boolean shortcutVisible(Context context, ShortcutInfo shortcutInfo, Set<
         return false;
     }
 
+    /**
+     * @return a new list which filters out shortcuts which come from apps which the user
+     * has excluded shortcuts for.
+     * Does not modify the input list.
+     */
+    private ArrayList<ShortcutPojo> filterOutExcludedApps(List<ShortcutPojo> allPojos, Context context) {
+        ArrayList<ShortcutPojo> result = new ArrayList<>();
+        Set<String> excludedApps = KissApplication.getApplication(context).getDataHandler().getExcludedShortcutApps();
+        for (ShortcutPojo pojo : allPojos) {
+            if (!excludedApps.contains(pojo.packageName)) {
```

Comment:
Wouln't this be better part of `shortcutVisible` method? At same place, where excludedApps are checked?
This would save something because `ShortcutPojo` wouldn't be created at all then.

Reply:
Great idea, I missed this. Changed